File: ruby/samples/get_crypto_book_bids.rb
# Load the gem
require 'intrinio-sdk'
 
# Setup authorization
Intrinio.configure do |config|
  config.api_key['api_key'] = 'YOUR API KEY'
end
 
crypto_api = Intrinio::CryptoApi.new
 
opts = { 
  pair: "btcusd", # String | Return the order book asks for the given Crypto Currency Pair.
  exchange: "gemini", # String | Return the order book asks for a Crypto Currency on the given Crypto Exchange.
  currency: "BTC" # String | Return the order book asks for the given Crypto Currency.
}

begin
  result = crypto_api.get_crypto_book_bids(opts)
  bids = result.bids
  puts "Crypto Currency Pair: #{result.pair.name}"
  puts "Crypto Exchange: #{result.exchange.name}"
  puts ""
  puts "BIDS"
  puts "-------------------------------------------------------------------------------------------"

  bids.each do |bid|
    puts ""
    puts "Price: #{bid.price}"
    puts "Size:  #{bid.size}"
  end
rescue Intrinio::ApiError => e
  puts "Exception when calling CryptoApi->get_crypto_book_bids: #{e}"
end

File: javascript-node/samples/getCryptoBookBids.js
var intrinioSDK = require('intrinio-sdk');
intrinioSDK.ApiClient.instance.authentications['ApiKeyAuth'].apiKey = "YOUR API KEY";
 
var cryptoAPI = new intrinioSDK.CryptoApi();
 
var opts = { 
  'pair': "btcusd", // String | Return the order book bids for the given Crypto Currency Pair.
  'exchange': "gemini", // String | Return the order book bids for a Crypto Currency on the given Crypto Exchange.
  'currency': "BTC" // String | Return the order book bids for the given Crypto Currency.
};
 
cryptoAPI.getCryptoBookBids(opts).then(function(data) {
  var bids = data.bids;
  var pair_name = data.pair.name;
  var exchange_name = data.exchange.name;

  console.log('Crypto Currency Pair: ' + pair_name);
  console.log('Crypto Exchange:      ' + exchange_name);
  console.log();
  console.log('---------------------------- BIDS ----------------------------');

  bids.forEach(function(bid) {
    console.log();
    console.log('Price: ' + bid.price);
    console.log('Size:  ' + bid.size);
  });
}, function(error) {
  console.error(error);
});
